internal/pkg/graph/graph.go

```diff
@@ -14,53 +16,53 @@ const (
 )
 
 // Graph represents a directed graph.
-type Graph struct {
-	nodes map[string]neighbors
+type Graph[V constraints.Ordered] struct {
```

Comment:
```suggestion
type Graph[T comparable] struct {
```

and then on `Edge`/`neighbors`